client/my-sites/themes/controller.js

```diff
@@ -49,27 +45,43 @@ function getProps( context ) {
 	};
 
 	return {
-		title,
 		tier,
 		search: context.query.s,
 		trackScrollPage: boundTrackScrollPage,
 		runClientAnalytics
 	};
-}
+};
 
-function makeElement( ThemesComponent, Head, store, props ) {
-	return(
-		<ReduxProvider store={ store }>
-			<Head title={ props.title } tier={ props.tier || 'all' }>
-				<ThemesComponent { ...omit( props, [ 'title', 'runClientAnalytics' ] ) } />
+const LoggedInHead = ( { children, context: { params: { tier, site_id: siteID } } } ) => (
+	<Head
+		title={ buildTitle(
+			i18n.translate( 'Themes', { textOnly: true } ),
+			{ siteID }
+		) }
+		tier={ tier || 'all' }>
+		{ children }
+	</Head>
+);
+
+// This is generic -- nothing themes-specific in here!
+function makeElement( Component, getProps, Head, sideEffects = function() {} ) {
+	// How do we handle the dispatch?
```

Comment:
@mcsf Any ideas? ^ (I mean stuff like https://github.com/Automattic/wp-calypso/blob/update/themes-functionalize/client/my-sites/themes/controller.js#L93-L96 which is currently present in each of our middlewares).